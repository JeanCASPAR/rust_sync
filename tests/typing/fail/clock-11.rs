use rustre::sync;

fn f(x: i64) -> i64 {
    x
}

sync! {
    #![pass(1)]
    
    node oui(b: bool) = ()
    where
        a : int = f(0 when b);
}

fn main() {}
