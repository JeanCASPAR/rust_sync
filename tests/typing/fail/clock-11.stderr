error: type mismatch: left has type `int`, right has type `int on b`
  --> tests/typing/fail/clock-11.rs:12:19
   |
12 |         a : int = f(0 when b);
   |                   ^^^^^^^^^^^
